src/module/dialogs/combat/CombatDefenseDialog.js

```diff
@@ -225,172 +440,293 @@ export class CombatDefenseDialog extends FormApplication {
+          spellCasting,
+          supShield,
+          defenderCombatMod
         }
+      });
 
-        const rolled = roll.total - magicProjection - (modifier ?? 0);
-
-        this.hooks.onDefense({
-          type: 'mystic',
-          values: {
-            modifier,
-            magicProjection,
-            spellGrade,
-            spellUsed,
-            at: at.final,
-            roll: rolled,
-            total: roll.total
-          }
-        });
-
-        this.modalData.defenseSent = true;
+      this.modalData.defenseSent = true;
 
-        this.render();
-      }
+      this.render();
     });
 
-    html.find('.send-psychic-defense').click(() => {
-      const { psychicProjection, psychicPotential, powerUsed, modifier } =
-        this.modalData.defender.psychic;
-      const { at } = this.modalData.defender.combat;
+    html.find('.send-psychic-defense').click(async () => {
+      const {
+        psychic: {
+          psychicPotential,
+          powerUsed,
+          modifier,
+          supernaturalShield: { shieldUsed, newShield }
+        },
+        combat: { at },
+        blindnessPen
+      } = this.modalData.defender;
+      const { i18n } = game;
+      const { psychicPowers } = this.defenderActor.system.psychic;
+      const { supernaturalShields } = this.defenderActor.system.combat;
+      let power,
+        fatigue,
+        supShield = { create: false },
+        newPsychicPotential;
+
+      const defenderCombatMod = {
+        modifier: { value: modifier, apply: true },
+        blindnessPen: { value: blindnessPen, apply: true }
+      };
+      const psychicProjection =
+        this.defenderActor.system.psychic.psychicProjection.imbalance.defensive.final
+          .value;
+      let combatModifier = 0;
+      for (const key in defenderCombatMod) {
+        combatModifier += defenderCombatMod[key]?.value ?? 0;
+      }
+      let formula = `1d100xa + ${psychicProjection} + ${combatModifier}`;
+      if (this.modalData.defender.withoutRoll) {
+        // Remove the dice from the formula
+        formula = formula.replace('1d100xa', '0');
+      }
+      if (this.defenderActor.system.psychic.psychicProjection.base.value >= 200) {
+        // Mastery reduces the fumble range
+        formula = formula.replace('xa', 'xamastery');
+      }
 
-      if (powerUsed) {
-        let formula = `1d100xa + ${psychicProjection} + ${modifier ?? 0}`;
-        if (this.modalData.defender.withoutRoll) {
-          // Remove the dice from the formula
-          formula = formula.replace('1d100xa', '0');
+      const psychicProjectionRoll = new ABFFoundryRoll(
+        formula,
+        this.defenderActor.system
+      );
+      psychicProjectionRoll.roll();
+      const rolled = psychicProjectionRoll.total - psychicProjection - combatModifier;
+
+      if (!newShield) {
+        if (!shieldUsed) {
+          return ui.notifications.warn(
+            i18n.localize('macros.combat.dialog.warning.supernaturalShield.psychic')
```

Comment:
Como en el caso místico:
```suggestion
            i18n.localize('macros.combat.dialog.warning.supernaturalShieldNotFound.psychic')
```

Reply:
Listo